src/app/legend/legend.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit } from '@angular/core';
+
+export const typeColors: readonly { color: string; tooltip: string }[] = Object.freeze([
```

Comment:
never used readonly as type, nice :P

please, put the type `{ color: string; tooltip: string}` in a separate variable and may inside a legend.model.ts file

Moreover, you can use `as const` at the end of the object instead of `Object.freeze()`.

Anyways, this seems written right :P

Reply:
As const and Object.freeze() have different goals as explained [here](https://stackoverflow.com/questions/33124058/object-freeze-vs-const).
Actually to make it really perfect I would need to apply Object.freeze() to all the contained objects too!

Reply:
mhm no

```typescript
// these should be the same
const x = { } as const;
const x = Object.freeze( {} );

const x = {}; // this is different, it is not const, the properties are not const
```

Reply:
That's not what I meant 😅 
As I said, they have two different uses.
Declaring a variable "const" means its reference can't be changed, but not its properties!
Object.freeze() instead prevents modifications of the object properties too, but stops at first level.

If you still don't believe me, try this code (decomment some lines to cover all cases):
```
const a: { changed?: boolean } = {} as const;
const b: { changed?: boolean } = Object.freeze({});
const c: { changed?: boolean } = Object.freeze({ changed: true });
const d: { changed?: { inner: boolean } } = Object.freeze({
  changed: {
    inner: true
  }
});

/* The following assignment fails because a is declared as const */
/* a = { changed: true }; */

/* But the properties are not const, so this assignment is legit */
a.changed = true;
console.log('a: ', a);

/* The following assignment fails because b is declared as const */
/* b = { changed: true }; */

/* Here the you can't add a new property because the object is frozen */
/* b.changed = true; */
console.log('b: ', b);

/* This assignment doesn't work because the property is read-only */
/* c.changed = false; */
console.log('c: ', c);

/* This works because the inner property is not frozen, even though the outer is */
if (d.changed) {
  d.changed.inner = false;
}
console.log('d: ', d);
```
Enjoy 😄

Reply:
I was convinced that even "as const" make the same of object.freeze()

Reply:
we could move this into another file if it is really shared with other files